testsuite/MDAnalysisTests/analysis/test_wbridge.py

```diff
@@ -0,0 +1,102 @@
+    2SOL     OW    3   0.300   0.000   0.000
+    2SOL    HW2    4   0.400   0.000   0.000
+    4ALA      O    5   0.600   0.000   0.000
+  1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        wb.run()
+        assert wb._timeseries == [[(1, 2, ('ALA', 1, 'H'), ('SOL', 2, 'OW'), 2.0, 180.0), (3, 4, ('SOL', 2, 'HW2'), ('ALA', 4, 'O'), 2.0, 180.0)]]
+
+    def test_acceptor_water_donor(self):
+        grofile = '''Test gro file
+5
+    1ALA      O    1   0.000   0.000   0.000
+    2SOL     OW    2   0.300   0.000   0.000
+    2SOL    HW1    3   0.200   0.000   0.000
+    4ALA      H    4   0.500   0.000   0.000
+    4ALA      N    5   0.600   0.000   0.000
+  1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        wb.run()
+        assert wb._timeseries == [[(2, 0, ('SOL', 2, 'HW1'), ('ALA', 1, 'O'), 2.0, 180.0), (3, 1, ('ALA', 4, 'H'), ('SOL', 2, 'OW'), 2.0, 180.0)]]
+
+    def test_donor_water_donor(self):
+        grofile = '''Test gro file
+5
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   0.300   0.000   0.000
+    4ALA      H    4   0.500   0.000   0.000
+    4ALA      N    5   0.600   0.000   0.000
+  1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        wb.run()
+        assert wb._timeseries == [[(1, 2, ('ALA', 1, 'H'), ('SOL', 2, 'OW'), 2.0, 180.0), (3, 2, ('ALA', 4, 'H'), ('SOL', 2, 'OW'), 2.0, 180.0)]]
+
+    def test_empty(self):
+        grofile = '''Test gro file
+5
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   3.000   0.000   0.000
+    4ALA      H    4   0.500   0.000   0.000
+    4ALA      N    5   0.600   0.000   0.000
+  1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein', 'protein')
+        wb.run()
+        assert wb._timeseries == [[]]
+
+    def test_same_selection(self):
+        '''
+        This test tests that if the selection 1 and selection 2 are both protein.
+        However, the protein only forms one hydrogen bond with the water.
+        This entry won't be included.
+        :return:
+        '''
+        grofile = '''Test gro file
+3
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   0.300   0.000   0.000
+  1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein', 'protein')
+        wb.run()
+        assert wb._timeseries == [[]]
+
+a = TestWaterBridgeAnalysis()
```

Comment:
These lines shouldn't be here, the test runner (in our case pytest) gathers the functions and runs them

Reply:
@richardjgowers Sorry, I should have deleted before committing.